server/vulnerabilities/io/metadata.go

```diff
@@ -0,0 +1,79 @@
+package io
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+)
+
+const (
+	mSRCFilePrefix              = "fleet_msrc_"
+	macOfficeReleaseNotesPrefix = "fleet_macoffice_release_notes_"
+	fileExt                     = "json"
+	dateLayout                  = "2006_01_02"
+)
+
+// MSRC Bulletins and other metadata files are published as assets to GH and copies are downloaded to the local FS. The file name
+// of those assets contain some useful information like the 'product name' and the date the asset was modified. This type
+// provides an abstration around the asset 'file name' to allow us to easy extract/compare the encoded info.
+type MetadataFileName struct {
+	prefix   string
+	filename string
+}
+
+func NewMSRCMetadata(filename string) MetadataFileName {
+	return MetadataFileName{prefix: mSRCFilePrefix, filename: filename}
+}
+
+func NewMacOfficeRelNotesMetadata(filename string) MetadataFileName {
+	return MetadataFileName{prefix: macOfficeReleaseNotesPrefix, filename: filename}
+}
+
+func (sbn MetadataFileName) date() (time.Time, error) {
+	parts := strings.Split(sbn.filename, "-")
+
+	if len(parts) != 2 {
+		return time.Now(), errors.New("invalid file name")
+	}
+	timeRaw := strings.TrimSuffix(parts[1], "."+fileExt)
+	return time.Parse(dateLayout, timeRaw)
+}
+
+func (sbn MetadataFileName) Before(other MetadataFileName) bool {
+	a, err := sbn.date()
+	if err != nil {
+		return false
```

Comment:
Better to return the value than ignoring the error.

If it's not possible for this to fail (e.g. because it's parsed before or something like that), then please add a comment.